### ISSUE TITLE
question of multiperiod hack to resolve time-shift of stream

### ISSUE THREAD

**Issue by kuznetcoff777** [google/shaka-player#1501]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**:

**What version of Shaka Player are you using?**:

**Can you reproduce the issue with our latest release version?**:

**Can you reproduce the issue with the latest code from `master`?**:

**Are you using the demo app or your own custom app?**:

**If custom app, can you reproduce the issue using our demo app?**:

**What browser and OS are you using?**:

**What are the manifest and license server URIs?**:

**What did you do?**

**What did you expect to happen?**

**What actually happened?**

Question is realted to case https://github.com/google/shaka-player/issues/899

Im attempting to get how can i hack my time-shift of a stream problem. I have an encoder that makes stream that is for some reasons go back for 3 seconds per 24 hours. I attempted to make another periods in mpd with period starts and presentationTimeOffset that takes into accont that shift but no luck. I get two kind of results:
-from some shift (it depends on presentation delay) player get 404 for segments( it feteches segmetns, that did not appear)
-time shows correctly, but real stream if out of real time (i see that for example newews segment is 1001, but player asks for 900, so real presentation delay is too huge)

Can you give me a hint how can i win this time-shift issue?

**Comment by joeyparrish** [comment]
Are you referring to encoder drift?  That's when the duration of what the encoder outputs is less than the time it took.  For example, every 24 hours, only 23.9 hours of content is output.

**Comment by kuznetcoff777** [comment]
Yes, it seems tobe this is it! It is great that you refactoring player!
Im attempting to resolve it via multi-period at the moment, but no luck at the moment.
When approximately this solution will be ready at your side?

**Comment by vaage** [comment]
@kuznetcoff777 Thank you for your patiences. Right now @joeyparrish is working on #999. While the initial approach was a larger-scaled solution (addressing many other problems), he is now working on identifying a smaller scale-scale solution that would resolve #999 in a smaller time window and then later bring in the larger-scale solution.

Last I talked with him, a strategy for this is still being developed as it #999 has been a difficult problem to navigate. Going forward I would suggest following #999 as it will allow you to get more direct updates on the progress.

If following #999 resolves all your needs, please feel free to close this issue.
